templates/vue/src/components/lightbox/H5PIframe.vue

```diff
@@ -175,35 +168,25 @@ export default {
     handleLoad() {
       this.$emit("is-loaded")
 
-      $("iframe").each(function() {
-        $(this)
-          .data("ratio", this.height / this.width)
-          // Remove the hardcoded width & height attributes
-          .removeAttr("width")
-          .removeAttr("height")
-      })
-      const setIframeDimensions = function() {
-        $("iframe").each(function() {
-          // Get the parent container's width
-          var width = $(this)
-            .parent()
-            .width()
-          var height = $(this)
-            .parent()
-            .height()
-          if (width * $(this).data("ratio") <= height) {
-            $(this)
-              .width(width)
-              .height(width * $(this).data("ratio"))
-          } else {
-            $(this)
-              .height(height)
-              .width(height / $(this).data("ratio"))
+      if (this.node.fullscreen) {
+        const frame = this.$refs.h5p
+        frame.removeAttribute("width")
+        frame.removeAttribute("height")
+        const setIframeDimensions = () => {
+          const { width, height } = frame.getBoundingClientRect()
+          const parentHeight = frame.parentNode.getBoundingClientRect().height
+          if (height > parentHeight) {
+            const ratio = height / width
+            if (ratio < 1) {
+              const newWidth = width * (parentHeight / height)
+              frame.style.width = newWidth + "px"
+              frame.style.height = parentHeight + "px"
+            }
```

Comment:
This works, but I don't necessarily like that I'm going into the DOM for this. I feel that there is a more elegant solution, but would require changing the html in other places of the lightbox. I was already going over 2 hours working on this so I thought I would leave it as is for now.